sdk/mixedreality/azure-mixedreality-authentication/azure/mixedreality/authentication/_client.py

```diff
@@ -0,0 +1,114 @@
+from ._shared.mixedreality_account_key_credential import MixedRealityAccountKeyCredential
+from ._utils import convert_to_access_token, generate_cv_base
+
+if TYPE_CHECKING:
+    # pylint: disable=unused-import,ungrouped-imports
+    from typing import Any, Union
+    from azure.core.credentials import TokenCredential
+    from azure.core.credentials import AccessToken
+
+
+class MixedRealityStsClient(object):
+    """ A client to interact with the Mixed Reality STS service.
+
+    :param str account_id:
+        The Mixed Reality service account identifier.
+    :param str account_domain:
+        The Mixed Reality service account domain.
+    :param Union[TokenCredential, AzureKeyCredential] credential:
+        The credential used to access the Mixed Reality service.
+    :keyword str custom_endpoint_url:
+        Override the Mixed Reality STS service endpoint.
+    """
+
+    def __init__(self, account_id, account_domain, credential, **kwargs):
+        # type: (str, str, Union[TokenCredential, AzureKeyCredential], Any) -> None
+        if not account_id:
+            raise ValueError("account_id can not be None")
+
+        if not account_domain:
+            raise ValueError("account_domain can not be None")
+
+        if not credential:
+            raise ValueError("credential can not be None")
+
+        self._account_id = account_id
+        self._account_domain = account_domain
+
+        if isinstance(credential, AzureKeyCredential):
+            credential = MixedRealityAccountKeyCredential(account_id, credential)
+
+        self._credential = credential
+
+        endpoint_url = kwargs.pop('custom_endpoint_url', construct_endpoint_url(account_domain))
+
+        try:
+            if not endpoint_url.lower().startswith('http'):
+                endpoint_url = "https://" + endpoint_url
+        except AttributeError:
+            raise ValueError("Host URL must be a string")
+
+        parsed_url = urlparse(endpoint_url.rstrip('/'))
+        if not parsed_url.netloc:
+            raise ValueError("Invalid URL: {}".format(endpoint_url))
+
+        self._endpoint_url = endpoint_url
+
+        authentication_policy = BearerTokenCredentialPolicy(credential, endpoint_url + '/.default')
+
+        self._client = MixedRealityStsRestClient(
+            base_url=endpoint_url,
+            authentication_policy=authentication_policy,
+            sdk_moniker=SDK_MONIKER,
+            **kwargs)
+
+    @distributed_trace
+    def get_token(self, **kwargs):
+        # type: (Any) -> AccessToken
+        """
+        Retrieve a token from the STS service for the specified account identifier asynchronously.
+        :return: Instance of azure.core.credentials.AccessToken - token and expiry date of it
+        :rtype: ~azure.core.credentials.AccessToken
+        """
+        token_request_options = TokenRequestOptions()
+        token_request_options.client_request_id = generate_cv_base()
```

Comment:
I don't know. I suppose it's possible. Our STS service (which is already GA) doesn't follow the typical method Azure recommends, so we pass our own. If I can tell the tooling to pass it as a different header value, then we could maybe use the azure-core tooling.

Reply:
Ok. I might need some more guidance on how to do this. What we're doing here is consistent with what we've done in all the other languages, so this will be a change.

Reply:
As discussed offline, we'll revisit this later when we have a solution across all the languages.